frame.go

```diff
@@ -769,14 +763,14 @@ func (f *framer) finishWrite() error {
 	length := len(f.wbuf) - f.headSize
 	f.setLength(length)
 
-	_, err := f.w.Write(f.wbuf)
```

Comment:
Yeah. There is only one write per framer (because we need to write the whole frame atomically).

We need to distinguish the errors returned from the network `Write` and other reasons like frame too big.
We could maybe rename `finishWrite` to `finish` and return the data to write from `finish`?

Reply:
I renamed `finishWrite` to `finish` and `frameWriter` interface to `frameBuilder`